compiler-plugin/src/main/kotlin/arrow/meta/quotes/classorobject/ObjectDeclaration.kt

```diff
@@ -0,0 +1,38 @@
+package arrow.meta.quotes.classorobject
+
+import arrow.meta.quotes.Scope
+import org.jetbrains.kotlin.psi.KtObjectDeclaration
+
+/**
+ * <code>""" object $name { $body }""".objectDeclaration</code>
+ *
+ * TODO add wrappers for certain boolean logic so that there are more options to create KtObjectDeclaration i.e. companion object
+ * @see [DefaultElementScope::companionObject]
+ *
+ * A template destructuring [Scope] for a [KtObjectDeclaration]. See below:
+ *
+ * ```kotlin:ank:silent
+ * import arrow.meta.Meta
+ * import arrow.meta.Plugin
+ * import arrow.meta.invoke
+ * import arrow.meta.quotes.Transform
+ * import arrow.meta.quotes.objectDeclaration
+ *
+ * val Meta.reformatObjectDeclaration: Plugin
+ *   get() =
+ *     "ReformatObjectDeclaration" {
+ *       meta(
+ *        objectDeclaration({ true }) { c ->
+ *          Transform.replace(
+ *            replacing = c,
+ *            newDeclaration = """ object $name { $body }""".`object`
```

Comment:
All templates in all examples should expose all available properties of the scope so the users can understand what a full declaration looks like and which parts they can use to compose the structure again. In this case, the examples are removing from the tree annotations, modifiers and supertypes. This may be true for other examples. All quote examples should include all members of the scope or the user may lose information as they copy and paste it from the docs into their projects.

Reply:
Yes, it is definitely true for other examples as well. I'm happy to add on to all that, but in order to create more examples, we need more exposed from PsiElements.

For example, we need additional information that we currently don't have in order to create a `companion object`.  This is what I meant by the Property example where we cannot choose to generate a val/var

Reply:
I think for the sake of this PR, it is okay to merge in basic examples and to continue working over all these elements by exposing what is necessary in order to create better examples.

Reply:
I'm going to push up an example of an element that has different kinds of examples for Parameter shortly so we can prototype future work (done this week but in different PRs) off that!

Reply:
Prototype can be observed in Parameter:

![Screen Shot 2019-11-26 at 3 50 42 PM](https://user-images.githubusercontent.com/6917498/69644324-ca98b180-1064-11ea-8f45-d2c5c4293427.png)